tests/framework/parser.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
```

Comment:
I think this isn't actually test-only code, it's actually useful more broadly. Maybe make a top-level parser package?

Alternatively, consider adding it directly to the `typed` package, then we'd never have to worry about import cycles.

Reply:
> Alternatively, consider adding it directly to the typed package, then we'd never have to worry about import cycles. 

Yeah, I like taht